src/editor/Gui.cpp

```diff
@@ -2617,6 +2829,153 @@ void Gui::drawAbout() {
+		ImGui::Checkbox("Optimize", &Optimize);
+		ImGui::Checkbox("No hull 2", &DeleteHull2);
+		ImGui::Checkbox("No ripent", &NoRipent);
+		ImGui::Checkbox("No script", &NoScript);
+		ImGui::Checkbox("Merge second map as model", &MergeSecondsMapAsModel);
+
+		if (ImGui::Button("Merge maps", ImVec2(120, 0)))
+		{
+			vector<Bsp*> maps;
+			for (auto const& s : g_app->mapRenderers)
+			{
+				maps.push_back(s->map);
+			}
+			if (maps.size() < 2)
+			{
+				logf("ERROR: at least 2 input maps are required\n");
+			}
+			else
+			{
+				for (int i = 0; i < maps.size(); i++) {
+					logf("Preprocessing %s:\n", maps[i]->name.c_str());
+					if (DeleteUnusedInfo)
+					{
+						logf("    Deleting unused data...\n");
+						STRUCTCOUNT removed = maps[i]->remove_unused_model_structures();
+						g_progress.clear();
+						removed.print_delete_stats(2);
+					}
+
+					if (DeleteHull2 || (Optimize && !maps[i]->has_hull2_ents())) {
+						logf("    Deleting hull 2...\n");
+						maps[i]->delete_hull(2, 1);
+						maps[i]->remove_unused_model_structures().print_delete_stats(2);
+					}
+
+					if (Optimize) {
+						logf("    Optmizing...\n");
+						maps[i]->delete_unused_hulls().print_delete_stats(2);
+					}
+
+					logf("\n");
+				}
+
+				BspMerger merger;
+				Bsp* result = merger.merge(maps, vec3(), Path, NoRipent, NoScript);
+
+				logf("\n");
+				if (result->isValid()) result->write(Path);
+				logf("\n");
+				result->print_info(false, 0, 0);
+
+				g_app->clearMaps();
+
+				if (fileExists(Path))
+				{
+					result = new Bsp(Path);
+					g_app->addMap(result);
+				}
+				else
+				{
+					logf("Error while map merge!\n");
+					g_app->addMap(new Bsp());
+				}
+			}
+			showMergeMapWidget = false;
+		}
+	}
+
+	ImGui::End();
+}
+
+void Gui::drawImportMapWidget() {
+	ImGui::SetNextWindowSize(ImVec2(500, 140), ImGuiCond_FirstUseEver);
+	ImGui::SetNextWindowSizeConstraints(ImVec2(500, 140), ImVec2(500, 140));
+	static char Path[256];
+	const char* title = "Import .bsp model as entity";
+
+	if (!showImportMapWidget_Import)
+	{
+		title = "Open new map";
+	}
+
+	if (ImGui::Begin(title, &showImportMapWidget)) {
+		ImGui::InputText(".bsp file", Path, 256);
+		if (ImGui::Button("Load", ImVec2(120, 0)))
+		{
+			fixupPath(Path, FIXUPPATH_SLASH::FIXUPPATH_SLASH_SKIP, FIXUPPATH_SLASH::FIXUPPATH_SLASH_SKIP);
+			if (fileExists(Path))
+			{
+				logf("Loading new map file from %s path.\n", Path);
+				showImportMapWidget = false;
+				if (!showImportMapWidget_Import)
+				{
+					g_app->addMap(new Bsp(Path));
+				}
+				else
+				{
+					if (g_app->mapRenderers.size() && g_app->mapRenderers[0]->map)
+					{
+						Bsp* model = new Bsp(Path);
+						if (!model->ents.size())
+						{
+							logf("Error! No worldspawn found!\n");
+							delete model;
+						}
+						else
+						{
+							Bsp* map = g_app->mapRenderers[0]->map;
+							logf("Adding cycler_sprite at pos -8000 -8000 -8000 to precache model.\n");

```

Comment:
why -8000? can it be created in front of the camera instead?

Reply:
> why -8000? can it be created in front of the camera instead?

Need to be hide it )) (Using only for precache .bsp model)

Reply:
I realized later this is for a feature I don't think should be here. Using the "create entity" option and setting the model to a BSP file is good enough for now. I know you can't see the model that way, but this is confusing implementation for an `Open` option. Eventually the entity renderer will be updated to show the MDL/BSP instead of a colored cube, which would let you see the same thing without having to `Open` a map to see what a model looks like for a specific entity.

Reply:
Yes It used for adding bsp model to map. (Auto create cycler_sprite and func_wall with selected models, without 'create entity' button)

Reply:
I can use func_breakable for precache and show model with real geometry.